packages/block-editor/src/components/global-styles/advanced-panel.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	TextareaControl,
+	Tooltip,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Icon, info } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { default as transformStyles } from '../../utils/transform-styles';
+
+export default function AdvancedPanel( {
```

Comment:
The idea is the component is equivalent to the "Advanced" section of the block inspector panel. It can contain things like custom class, anchor, ... and any "common" advanced setting.

The issue here is kind of similar to the "filters" panel where we only have duotone and the "effects" panel where we only have "box shadow" so far.

Reply:
Got it. So this panel would replace the block inspector Advanced section, once it's expanded to handle custom classes etc.?

Reply:
That's my hope but I'm not entirely sure. Visually at least, it helps creates a relation between the two panels (local and global ones)

Reply:
> I'm not sure I understand. Visually the custom CSS panel in this branch looks the same as in trunk; they both use similar fonts and spacing as the block inspector panels.
> Is the idea to potentially add custom CSS to the block inspector so it can be set on individual blocks? Right now the logic in this AdvancedPanel is pretty specific to custom CSS.

My logic is the following. Within "inspector controls" we have "panels" and each group has "controls".

 - Typography panel has a dozen of controls (font size, font family, line height...)
 - Color panel has some controls too (background, text, link, ...)
 - Dimensions panel has some controls too (height, width, padding, margin...)
 - Effects panel has some controls (only box shadows now)
 - Filters panel has some controls (only duotone now)
 - Advanced panel has some controls too (custom CSS). It turns out that for this one right now, custom CSS is global style only and the rest are block inspector only, that said, it was the same for "duotone" when I implemented the panel, and I believe it's the same for the effects panel too. But it doesn't mean that it will always be the case, custom CSS can be added to block inspector at any point. 
 
Does that make sense?

Reply:
> custom CSS can be added to block inspector at any point

If that's the case then it makes sense to have this here. Though if we ever add more global styles controls here, it might be better to change the name of the site editor component from `CustomCSSControl` to something more generic too.

Reply:
> better to change the name of the site editor component from CustomCSSControl to something more generic too.

I actually agree with this, It should be AdvancedPanel too there too. It's just that I got confused, I thought you were discussing the ones in the block editor package. I'll make the update.

Reply:
Ok, so I just took a look again. I think the issue right now with the naming is that we access the panel by clicking "Additional CSS" button and going into an "Custom CSS screen" which contains this component.

So if I update `CustomCSSControl` to `AdvancedPanel`, I'll be just moving the issue to another place because the CustomCSSScreen will include AdvancedPanel. (so same issue).

I think this is just a temporary issue though that will be solved when we unify the whole block sidebar in #49428 since we'll just be rendering all the panels inline exactly like the block inspector. So I think we shouldn't be doing anything here for now until the other PR.

Reply:
That makes sense!  Another thing that would be good to look at in a follow up would be adding a "reset" button that wipes any user-added CSS and restores the theme CSS if there is any.